events/events_network.c

```diff
@@ -292,30 +323,47 @@ struct eventrec *
 events_network_get(void)
 {
 	struct eventrec * r;
-	size_t nfds = socketlist_getsize(S);
 
 	/* We haven't found any events yet. */
 	r = NULL;
 
+	/* We have no fds, thus we have no events. */
+	if (fds == NULL)
+		return (r);
+
 	/* Scan through the fd sets looking for ready sockets. */
 	for (; fdscanpos < nfds; fdscanpos++) {
+		struct pollfd * current = &fds[fdscanpos];
+
+		/* File descriptor was closed, ignore it in future polls. */
+		if (current->revents & POLLNVAL) {
+			current->fd = -1;
```

Comment:
Changing it to an assert broke stuff; there isn't anywhere else that handles "file descriptor was closed".